### ISSUE TITLE
Specified cast is not valid when attempting single file restore

### ISSUE THREAD

**Issue by autumnwalker** [duplicati/duplicati#2246]
I have:
- [x] searched open and closed issues for duplicates

----------------------------------------

### Version info
**Duplicati Version:** 2.0.1.39_canary_2017-01-08
**Operating System:** Ubuntu 16.04
**Backend:** Amazon Cloud Drive

### Bug description
When attempting to restore a single file from a completed backup the WebUI indicates that the file has been restored successfully,  but I get an error saying "Specified cast is not valid" and the file is not actually restored.

### Steps to reproduce
- complete backup
- attempt to restore individual file

### Screenshots
![restore_error](https://cloud.githubusercontent.com/assets/11489678/21815863/773f26c0-d734-11e6-8f74-2096a78d2d89.jpg)

### debug log
```
System.InvalidCastException: Specified cast is not valid.
  at Mono.Data.Sqlite.SqliteDataReader.VerifyType (Int32 i, DbType typ) <0x40cdac00 + 0x001ab> in <filename unknown>:0 
  at Mono.Data.Sqlite.SqliteDataReader.GetString (Int32 i) <0x415dc130 + 0x00077> in <filename unknown>:0 
  at Duplicati.Library.Main.Database.LocalRestoreDatabase+<GetFilesToRestore>c__Iterator1.MoveNext () <0x4162f630 + 0x00230> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.RestoreHandler.DoRun (Duplicati.Library.Main.Database.LocalDatabase dbparent, IFilter filter, Duplicati.Library.Main.RestoreResults result) <0x41617090 + 0x017bb> in <filename unknown>:0 
  at Duplicati.Library.Main.Operation.RestoreHandler.Run (System.String[] paths, IFilter filter) <0x41616280 + 0x002cb> in <filename unknown>:0 
  at Duplicati.Library.Main.Controller+<Restore>c__AnonStorey1.<>m__0 (Duplicati.Library.Main.RestoreResults result) <0x41615f40 + 0x0007f> in <filename unknown>:0 
  at Duplicati.Library.Main.Controller.RunAction[T] (Duplicati.Library.Main.T result, System.String[]& paths, IFilter& filter, System.Action`1 method) <0x4154b710 + 0x001dd> in <filename unknown>:0
```

**Comment by autumnwalker** [comment]
FYI - I tried the restore again, but this time manually specified the folder path to match the "original location" and it worked.

Issue appears to be isolated to when you tell Duplicati to restore files to "original location".

**Comment by kenkendk** [comment]
I think this was introduced by the cross-os fixes in 2.0.1.39
